core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -2073,19 +2118,18 @@ func (o *evmTxStore) FindConfirmedTxesReceipts(ctx context.Context, finalizedBlo
 	var cancel context.CancelFunc
 	ctx, cancel = o.stopCh.Ctx(ctx)
 	defer cancel()
-	err = o.Transact(ctx, true, func(orm *evmTxStore) error {
-		sql := `SELECT evm.receipts.* FROM evm.receipts
+
+	// note the receipts are partially loaded for performance reason
+	query := `SELECT evm.receipts.id, evm.receipts.tx_hash, evm.receipts.block_hash, evm.receipts.block_number FROM evm.receipts
 		INNER JOIN evm.tx_attempts ON evm.tx_attempts.hash = evm.receipts.tx_hash
 		INNER JOIN evm.txes ON evm.txes.id = evm.tx_attempts.eth_tx_id
 		WHERE evm.txes.state = 'confirmed' AND evm.receipts.block_number <= $1 AND evm.txes.evm_chain_id = $2`
-		var dbReceipts []DbReceipt
-		err = o.q.SelectContext(ctx, &dbReceipts, sql, finalizedBlockNum, chainID.String())
-		if len(dbReceipts) == 0 {
-			return nil
-		}
+	var dbReceipts []DbReceipt
+	err = o.q.SelectContext(ctx, &dbReceipts, query, finalizedBlockNum, chainID.String())
+	if len(dbReceipts) != 0 {
```

Comment:
No need for this, you can do:
```suggestion
	err = o.q.SelectContext(ctx, &receipts, query, finalizedBlockNum, chainID.String())
	return receipts, err
```